libbeat/publisher/queue/diskqueue/segments.go

```diff
@@ -470,6 +470,35 @@ func (segments *diskQueueSegments) sizeOnDisk() uint64 {
 	return total
 }
 
+// Iterates through all segment types to find the oldest ID in the queue
+func (segments *diskQueueSegments) oldestIDOnDisk() segmentID {
+	// not all segment types are pre-sorted, hence us appending some, taking [0] from others
```

Comment:
It's not clear to me what this means, in particular what does "taking [0] from others mean" here?